queries.sql

```diff
@@ -29,3 +29,27 @@ WHERE name != 'Gabumon';
 -- Find all animals with a weight between 10.4kg and 17.3kg (including the animals with the weights that equals precisely 10.4kg or 17.3kg)
 SELECT * FROM animals
 WHERE weight_kg >= 10.4 and weight_kg <= 17.3;
+
+-------------------------------------------------------------------
+
+-- How many animals are there?
+SELECT COUNT(*) FROM animals;
+
+-- How many animals have never tried to escape?
+SELECT COUNT(*) FROM animals
+WHERE escape_attempts = 0;
+
+-- What is the average weight of animals?
+SELECT AVG(weight_kg) FROM animals;
+
+-- Who escapes the most, neutered or not neutered animals?
+SELECT neutered, AVG(escape_attempts) FROM animals
+GROUP BY neutered;
+-- What is the minimum and maximum weight of each type of animal?
+SELECT species, MIN(weight_kg), MAX(weight_kg) FROM animals
+GROUP BY species;
+
+-- What is the average number of escape attempts per animal type of those born between 1990 and 2000?
+SELECT species, AVG(escape_attempts) FROM animals
+WHERE date_of_birth BETWEEN '19900101' AND '20210101'
+GROUP BY species;
```

Comment:
- [ ] The query should be between 1990 and 2000. Please kindly ensure the date is correct

Reply:
@AnselemOdims Thank you very much for leaving the comment.
The BETWEEN in SQL works a bit differently than the expected date. Kindly, keep in mind that the first date is inclusive, but the second is exclusive, as they effectively are '1990/01/01 00:00:00' and '2001/01/01 00:00:00'.
If I change this it will not be precise.
Can you please advise?